tests/transforms/test_split_non_commuting.py

```diff
@@ -0,0 +1,241 @@
+    def circuit(params):
+        qml.RZ(params[0], wires=1)
+        qml.RZ(params[1], wires=2)
+        return (
+            qml.expval(qml.PauliX(0) @ qml.PauliX(1)),
+            qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
+            qml.expval(qml.PauliX(0)),
+        )
+
+    params = pnp.array([np.pi, 0.0])
+    res = circuit(params)
+    grad = qml.jacobian(circuit)(params)
+    assert all(np.isclose(res, [0, 1, 0]))
+    assert all(np.isclose(grad, np.zeros((3, 2))).flatten())
+
+
+# TODO: Currently not possible to jit multiple expvals
+@pytest.mark.jax
+def test_split_with_jax():
+    """Test that results after splitting are still differentiable with jax"""
+
+    import jax
+    import jax.numpy as jnp
+
+    dev = qml.device("default.qubit.jax", wires=3)
+
+    @qml.qnode(dev, interface="jax")
+    def circuit(params):
+        qml.RZ(params[0], wires=1)
+        qml.RZ(params[1], wires=2)
+        return (
+            qml.expval(qml.PauliX(0) @ qml.PauliX(1)),
+            qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
+            qml.expval(qml.PauliX(0)),
+        )
+
+    params = jnp.array([np.pi, 0.0])
+    res = circuit(params)
+    grad = jax.jacobian(circuit)(params)
+    assert all(np.isclose(res, [0, 1, 0]))
+    assert all(np.isclose(grad, np.zeros((3, 2))).flatten())
+
+
+@pytest.mark.torch
+def test_split_with_torch():
+    """Test that results after splitting are still differentiable with torch"""
+
+    import torch
+    from torch.autograd.functional import jacobian
+
+    dev = qml.device("default.qubit.torch", wires=3)
+
+    @qml.qnode(dev, interface="torch")
+    def circuit(params):
+        qml.RZ(params[0], wires=1)
+        qml.RZ(params[1], wires=2)
+        return (
+            qml.expval(qml.PauliX(0) @ qml.PauliX(1)),
+            qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
+            qml.expval(qml.PauliX(0)),
+        )
+
+    params = torch.tensor([np.pi, 0], requires_grad=True)
+    res = circuit(params)
+    grad = jacobian(lambda params: circuit(params), (params))
+    assert all(np.isclose(res.detach().numpy(), [0, 1, 0]))
+    assert all(np.isclose(grad.detach().numpy(), np.zeros((3, 2))).flatten())
+
+
+@pytest.mark.tf
+def test_split_with_tf():
+    """Test that results after splitting are still differentiable with tf"""
+
+    import tensorflow as tf
+    dev = qml.device("default.qubit.tf", wires=3)
+
+    @qml.qnode(dev, interface="tf")
+    def circuit(params):
+        qml.RZ(params[0], wires=1)
+        qml.RZ(params[1], wires=2)
+        return (
+            qml.expval(qml.PauliX(0) @ qml.PauliX(1)),
+            qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
+            qml.expval(qml.PauliX(0)),
+        )
+
+    params = tf.Variable([np.pi, 0])
+    res = circuit(params)
+    with tf.GradientTape() as tape:
+        loss = circuit(params)
+
+    grad = tape.jacobian(loss, params)
+    assert all(np.isclose(res, [0, 1, 0]))
+    assert all(np.isclose(grad, np.zeros((3, 2))).flatten())
```

Comment:
Very nice testing @Qottmann! I must admit I am curious why the gradients are always zero. Is this the expected gradient?

Since a gradient of zero can often indicate issues (e.g., backpropagation failed), it would be nice to ensure that the circuit above has a non-zero gradient value, and if not, tweak it so that it does

Reply:
You had me staring at the Bloch sphere again for a while there!

The PauliZ one is zero because rotating around the Z axis does not do anything when you are already on the Z axis. 

The PauliX ones are zero because they are sitting at the extremum (it also matters where you evaluate the gradient!).

But I agree that those are not the best tests and some numerical values would make sense. I just didnt want to use my execution as the test but actually since there is no entanglement I can just execute them individually (so without grouping into commuting groups) and use those results.